### ISSUE TITLE
Reproducing paper's results

### ISSUE THREAD

**Issue by frezaeix** [ChrisAllenMing/GPA-detection#8]
Hey! thank you for sharing your code,

I have a question regarding aligning features only for RPN or only for RCNN stage (sim to city); I trained 2 models with following commands:

Only for RPN
train_GPA.py --dataset sim10k --tgt_dataset city --model_config gcn_adapt --mode gcn_adapt --rpn_mode gcn_adapt --net res50 --bs 3 --nw 1 --epochs 20 --lr 0.001 --lr_decay_step 16 --lr_decay_gamma 0.1 --warm_up 200 **--da_weight 0.0 --rpn_da_weight 1.0** --cuda

Only for RCNN:
train_GPA.py --dataset sim10k --tgt_dataset city --model_config gcn_adapt --mode gcn_adapt --rpn_mode gcn_adapt --net res50 --bs 3 --nw 1 --epochs 20 --lr 0.001 --lr_decay_step 16 --lr_decay_gamma 0.1 --warm_up 200 **--da_weight 1.0 --rpn_da_weight 0.0** --cuda

For the first experiment, I got ~30 mAP and for the second one I got 40, can you help me with it?
did I do anything wrong? I also checked results after 10 epoch with lr_decay_step  of 6, they are like these numbers.

I should mention that I could get around 46 mAP by enabling both of them.

**Comment by ChrisAllenMing** [comment]
Hey! Thanks for your questions, these results may caused by following reasons:

- First, in our experiments, we use 2 GPUs and set 'batch size' as 6 (3 images on each GPU). We found that larger batch size benefits model's performance.
- Second, we found that the peak performance on target domain is commonly achieved in the 4th or 5th epoch, and it is better to test the models of all epoches using 'iterative_test.py'. Also, in the sim10k -> city task, we actually didn't use the learning rate decay scheme, i.e. fixed learning rate along training.
- We checked our code and found a flaw that the 'positive_ratio' parameter is inconsistent in training and test. We have fixed this parameter as 0.25, and you can download the code of newest version.

**Comment by frezaeix** [comment]
Hey, thanks for your response.  I will try again with these new points.

Do you think this minor change and using a higher batch size will improve the performance by 15 percent for the first experiment(Only for RPN)?  I think there is a huge gap here.

**Comment by Leesoon1984** [comment]
我有一个疑问，就是在迁移学习训练过程中，在目标域上没有标注的情况下，是如何判断模型在目标域上达到最优的？最优的模型，是选择测试后最大的那一个吗？

**Comment by Feobi1999** [comment]
> 我有一个疑问，就是在迁移学习训练过程中，在目标域上没有标注的情况下，是如何判断模型在目标域上达到最优的？最优的模型，是选择测试后最大的那一个吗？

我也有这个疑问，因为之前训baseline的时候看到损失还在降，拿sim—cityscape 举例子，在cityscape上的测试已经达到最优了

**Comment by zpge** [comment]
想问一个问题，域适应模型与直接在目标域上进行训练，mAP差别有多大？比如sim10k 到city的域适应，与直接在city上训练，结果对比如何？文章里貌似只有sim10k 到city的域适应与在source上的比较。

**Comment by weix-liu** [comment]
确实，这个batch size不知道有多大影响
